### PR TITLE
Pass quarkus args to dev mode gradle task

### LINKED ISSUE #28443: quarkus.args not work with gradle and quarkus 2.13.0
### Describe the bug

I try to write my own main like that:
```
@QuarkusMain
public class Main {
    public static void main(String... args) {
        System.out.println(args.length);
    }
}
```

And execute it using `quarkus cli` like that 
> quarkus dev --verbose -- param
or
> quarkus dev --verbose -Dquarkus.args="param"

### Expected behavior

The main is executed and display "1"

### Actual behavior

The main is executed and display "0"

### How to Reproduce?

_No response_

### Output of `uname -a` or `ver`

Darwin --- 21.5.0 Darwin Kernel Version 21.5.0: Tue Apr 26 21:08:22 PDT 2022; root:xnu-8020.121.3~4/RELEASE_X86_64 x86_64 i386 MacBookPro16,1 Darwin

### Output of `java -version`

1.17

### GraalVM version (if different from Java)

2.13.0

### Quarkus version or git rev

_No response_

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

7.4.2

### Additional information

In verbose mode I can see that parameter is set on quarkus.args but seem not used:

> gradlew --console=rich -DquarkusRegistryClient=true quarkusDev -Dquarkus.args='param' -Dio.quarkus.devmode-args=/var/folders/4q/z24jsww13cv73x6jqyqlj_880000gn/T/quarkus-dev2169293274555449898.txt

### PR DESCRIPTION
**Pull request by jacobdotcosta**
~Tried to build a test case for this fix but wasn't able to do so as `dev` mode doesn't end until you either press `CTRL+C` or `q` (quit) on the console .~

Tested the fix using the [jacobdotcosta/quarkus-issue-28443](https://github.com/jacobdotcosta/quarkus-issue-28443/) project.

Closes #28443

### CODE REVIEW

**Review by glefloch — APPROVED** [review]
Thanks for this @jacobdotcosta. I will try to push a tests I worked on.

**Comment by aloubyansky** [comment]
@jacobdotcosta Have you seen how dev mode is tested in https://github.com/quarkusio/quarkus/tree/main/integration-tests/gradle/src/test/java/io/quarkus/gradle/devmode?

**Comment by jacobdotcosta** [comment]
> @jacobdotcosta Have you seen how dev mode is tested in https://github.com/quarkusio/quarkus/tree/main/integration-tests/gradle/src/test/java/io/quarkus/gradle/devmode?

No, I didn't encounter those tests when I looked for them 🤦🏽 . I'll take a look at them now. Thank you @aloubyansky .

**Comment by jacobdotcosta** [comment]
I've implemented a test case for this but it's failing and I need your help with it. Let's hope I can explain myself correctly.

I wasn't sure of a way to check the `main` method information result as there is no access to the logs in the current `io.quarkus.gradle.devmode.QuarkusDevGradleTestBase` class implementation. The way I did it was by using 2 static variables in the `org.acme.EntryPoint`  class which is implementing the `main` method.

The problem I'm having is that the result of parsing the `-Dquarkus.args="param1=1 param2=2"` in the Java integration tests is different from calling the dev mode from the command line, let me explain.

When I launch the test with the `-Dquarkus.args="param1=1 param2=2"` parameter I expect that it will be translated to an array of 2 String as it happens when I call the dev mode from the command line. The problem is that the testing dev mode isn't making this parsing so the test is failing.

The test output is the following.

```
...
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running io.quarkus.gradle.devmode.BasicJavaMainApplicationModuleDevModeTest
$ /...../quarkus/integration-tests/gradle/./gradlew -D=maven.repo.local=/home/xxx/.m2/repository -Dorg.gradle.console=plain --stacktrace --info clean quarkusDev -Dquarkus.args="param1=1 param2=2"
getHttpResponse: [param1=1 param2=2]
getHttpResponse: 1
BELOW IS THE CAPTURED LOGGING OF THE FAILED GRADLE TEST PROJECT BUILD
...
Press [h] for more options>
Tests paused
Press [r] to resume testing, [h] for more options>
Press [r] to resume testing, [o] Toggle test output, [h] for more options>
args.length: 1
args: [param1=1 param2=2]
__  ____  __  _____   ___  __ ____  ______ 
 --/ __ \/ / / / _ | / _ \/ //_/ / / / __/ 
 -/ /_/ / /_/ / __ |/ , _/ ,< / /_/ /\ \   
--\___\_\____/_/ |_/_/|_/_/|_|\____/___/   
2022-11-24 14:05:10,001 INFO  [io.quarkus] (Quarkus Main Thread) code-with-quarkus unspecified on JVM (powered by Quarkus 999-SNAPSHOT) started in 1.988s. Listening on: http://localhost:8080
2022-11-24 14:05:10,018 INFO  [io.quarkus] (Quarkus Main Thread) Profile dev activated. Live Coding activated.
2022-11-24 14:05:10,019 INFO  [io.quarkus] (Quarkus Main Thread) Installed features: [cdi, resteasy, smallrye-context-propagation, vertx]

[ERROR] Tests run: 1, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 10.242 s <<< FAILURE! - in io.quarkus.gradle.devmode.BasicJavaMainApplicationModuleDevModeTest
[ERROR] io.quarkus.gradle.devmode.BasicJavaMainApplicationModuleDevModeTest.main  Time elapsed: 9.959 s  <<< FAILURE!
java.lang.AssertionError: 

Expecting actual:
  "1"
to contain:
  "2" 
```

But if I execute the `/...../quarkus/integration-tests/gradle/./gradlew -D=maven.repo.local=/home/ajc102/.m2/repository -Dorg.gradle.console=plain --stacktrace --info clean quarkusDev -Dquarkus.args="param1=1 param2=2"` from the built project (by temporary removing the line that deletes the temporary projectDir `FileUtils.deleteQuietly(projectDir);`), I get the expected result.

```
Press [h] for more options>
Tests paused
Press [r] to resume testing, [h] for more options>
Press [r] to resume testing, [o] Toggle test output, [h] for more options>
args.length: 2
args: [param1=1, param2=2]
__  ____  __  _____   ___  __ ____  ______ 
 --/ __ \/ / / / _ | / _ \/ //_/ / / / __/ 
 -/ /_/ / /_/ / __ |/ , _/ ,< / /_/ /\ \   
--\___\_\____/_/ |_/_/|_/_/|_|\____/___/   
2022-11-24 13:53:32,811 INFO  [io.quarkus] (Quarkus Main Thread) code-with-quarkus unspecified on JVM (powered by Quarkus 999-SNAPSHOT) started in 1.469s. Listening on: http://localhost:8080
2022-11-24 13:53:32,825 INFO  [io.quarkus] (Quarkus Main Thread) Profile dev activated. Live Coding activated.
2022-11-24 13:53:32,825 INFO  [io.quarkus] (Quarkus Main Thread) Installed features: [cdi, resteasy, smallrye-context-propagation, vertx]
```

Could you give me a hand with this @glefloch @aloubyansky

**Comment by aloubyansky** [comment]
@jacobdotcosta you can see how the log is printed in `QuarkusDevGradleTestBase`. You could try reading the `command-output.log` or add some changes to expose the log to the test, for example as a post test assertion.

As to the parsing, you could try checking the value of `quarkus.args` in `QuarkusDev`, it probably somehow gets quotes and that's an issue.

**Review by gsmet — CHANGES_REQUESTED** [review]
Just a note that this will need a squash before merging.

**Comment by gsmet** [comment]
I rebased and force pushed. There's a good chance the test failure is not related but let's get another CI run.

**Comment by jacobdotcosta** [comment]
Thank you @gsmet , I was testing doing the squash as I never did it before. 😨 

The test error is something that I haven't been able to find yet the reason. If I comment the delete temporary project delete and launch the same `gradlew` command it works. But the one executed inside the unit test parses the arguments wrong as 1 argument only.

**Comment by jacobdotcosta** [comment]
The MacOS tests have failed. I can see a `devtools/gradle/gradlew: No such file or directory` but I'm not sure what is causing this.

```
...
2023-01-04T15:29:22.9663660Z Starting machine "podman-machine-default"
2023-01-04T15:29:23.4807730Z Waiting for VM ...
2023-01-04T15:29:38.8691100Z Mounting volume... /Volumes/actions-files/githubactions:/Volumes/actions-files/githubactions
2023-01-04T15:29:39.3572700Z Error: exit status 255
2023-01-04T15:29:39.3588410Z ##[error]Process completed with exit code 125.
2023-01-04T15:29:39.3763220Z ##[group]Run devtools/gradle/gradlew --stop && rm -rf devtools/gradle/gradle-extension-plugin/build/tmp
2023-01-04T15:29:39.3763550Z [36;1mdevtools/gradle/gradlew --stop && rm -rf devtools/gradle/gradle-extension-plugin/build/tmp[0m
2023-01-04T15:29:39.3779080Z shell: /bin/bash --noprofile --norc -e -o pipefail {0}
2023-01-04T15:29:39.3779250Z env:
2023-01-04T15:29:39.3779370Z   LANG: en_US.UTF-8
2023-01-04T15:29:39.3779550Z   COMMON_MAVEN_ARGS: -e -B --settings .github/mvn-settings.xml --fail-at-end
2023-01-04T15:29:39.3779970Z   NATIVE_TEST_MAVEN_ARGS: -Dtest-containers -Dstart-containers -Dquarkus.native.native-image-xmx=5g -Dnative -Dnative.surefire.skip -Dformat.skip -Dno-descriptor-tests install -DskipDocs
...
2023-01-04T15:29:39.3781250Z ##[endgroup]
2023-01-04T15:29:39.4106020Z /Volumes/actions-files/githubactions/actions-runner/_work/_temp/fad6c767-635b-422d-9804-50b6c61f4a52.sh: line 1: devtools/gradle/gradlew: No such file or directory
2023-01-04T15:29:39.4109370Z ##[error]Process completed with exit code 1.
```

File: integration-tests/gradle/src/test/java/io/quarkus/gradle/devmode/BasicJavaMainApplicationModuleDevModeTest.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+import java.io.File;
+import java.io.RandomAccessFile;
+import java.nio.file.Files;
+import java.util.Arrays;
+import java.util.concurrent.TimeUnit;
+
+import org.junit.jupiter.api.condition.DisabledOnOs;
+import org.junit.jupiter.api.condition.OS;
+
+@DisabledOnOs(OS.WINDOWS)
```
Comment (aloubyansky) [inline integration-tests/gradle/src/test/java/io/quarkus/gradle/devmode/BasicJavaMainApplicationModuleDevModeTest.java:16]: @jacobdotcosta, could you please add a comment why it is disabled on Windows?

File: integration-tests/gradle/src/test/java/io/quarkus/gradle/devmode/BasicJavaMainApplicationModuleDevModeTest.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+        return "basic-java-main-application-project";
+    }
+
+    @Override
+    protected String[] buildArguments() {
+        return new String[] { "clean", "quarkusDev", "-s", "-Dquarkus.args=param1=1 param2=2" };
+    }
+
+    protected void testDevMode() throws Exception {
+        System.out.println("testDevMode()...");
```
Comment (aloubyansky) [inline integration-tests/gradle/src/test/java/io/quarkus/gradle/devmode/BasicJavaMainApplicationModuleDevModeTest.java:31]: This println could be removed

**Comment by aloubyansky** [comment]
@jacobdotcosta please squash your commits into one.

File: integration-tests/gradle/src/test/java/io/quarkus/gradle/devmode/BasicJavaMainApplicationModuleDevModeTest.java (hunk 1 above)
Comment (jacobdotcosta) [reply integration-tests/gradle/src/test/java/io/quarkus/gradle/devmode/BasicJavaMainApplicationModuleDevModeTest.java:16]: My test was copy/pasted from `BasicJavaMainApplicationModuleDevModeTest.java` which had that tag. I'll remove it and test again.

File: integration-tests/gradle/src/test/java/io/quarkus/gradle/devmode/BasicJavaMainApplicationModuleDevModeTest.java (hunk 2 above)
Comment (jacobdotcosta) [reply integration-tests/gradle/src/test/java/io/quarkus/gradle/devmode/BasicJavaMainApplicationModuleDevModeTest.java]: fixed

**Comment by aloubyansky** [comment]
@gsmet could you please give it another look?

**Review by gsmet — APPROVED** [review]

**Comment by gsmet** [comment]
Thanks a lot and sorry for the delay!

**Comment by gsmet** [comment]
Apparently the test is failing in main. I will revert it for now and open a new PR tomorrow.

**Comment by jacobdotcosta** [comment]
I'll take a look.

**Comment by gsmet** [comment]
@jacobdotcosta OK. Let me know if you need anything from us.

**Pull request merged**